File: tests/algorithms/x/Java/project_euler/problem_007/sol2.java
public class Main {
    static int ans;

    static boolean is_prime(int number) {
        if (number > 1 && number < 4) {
            return true;
        } else         if (number < 2 || Math.floorMod(number, 2) == 0 || Math.floorMod(number, 3) == 0) {
            return false;
        }
        int i = 5;
        while (i * i <= number) {
            if (Math.floorMod(number, i) == 0 || Math.floorMod(number, (i + 2)) == 0) {
                return false;
            }
            i = i + 6;
        }
        return true;
    }

    static int solution(int nth) {
        if (nth <= 0) {
            throw new RuntimeException(String.valueOf("Parameter nth must be greater than or equal to one."));
        }
        int[] primes = ((int[])(new int[]{}));
        int num = 2;
        while (primes.length < nth) {
            if (((Boolean)(is_prime(num)))) {
                primes = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(primes), java.util.stream.IntStream.of(num)).toArray()));
            }
            num = num + 1;
        }
        return primes[primes.length - 1];
    }
    public static void main(String[] args) {
        ans = solution(10001);
        System.out.println("solution(10001) = " + _p(ans));
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/PHP/project_euler/problem_007/sol2.php
<?php
ini_set('memory_limit', '-1');
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
function _panic($msg) {
    fwrite(STDERR, strval($msg));
    exit(1);
}
function is_prime($number) {
  global $ans;
  if ($number > 1 && $number < 4) {
  return true;
} else {
  if ($number < 2 || $number % 2 == 0 || $number % 3 == 0) {
  return false;
};
}
  $i = 5;
  while ($i * $i <= $number) {
  if ($number % $i == 0 || $number % ($i + 2) == 0) {
  return false;
}
  $i = $i + 6;
};
  return true;
}
function solution($nth) {
  global $ans;
  if ($nth <= 0) {
  _panic('Parameter nth must be greater than or equal to one.');
}
  $primes = [];
  $num = 2;
  while (count($primes) < $nth) {
  if (is_prime($num)) {
  $primes = _append($primes, $num);
}
  $num = $num + 1;
};
  return $primes[count($primes) - 1];
}
$ans = solution(10001);
echo rtrim('solution(10001) = ' . _str($ans)), PHP_EOL;
